### ISSUE TITLE
Support  html view for  WebUI

### ISSUE THREAD

**Issue by kylixs** [alibaba/arthas#1168]
At present, a simple WebUI is implemented based on http api, which can only display json data and is not suitable for human reading. We need a text / html view to better display the results of command interaction. It expects to provide a modern operating experience, richer functions, and more flexible scalability.

**Comment by kylixs** [comment]
For details https://github.com/alibaba/arthas/pull/1171

**Comment by sjha2048** [comment]
@kylixs how do I test this pull request on my local machine?

**Comment by kylixs** [comment]
> @kylixs how do I test this pull request on my local machine?

This http api branch will be merged into arthas in the near future, please use the PR branch for testing now.

**Comment by arteevraina** [comment]
Hello @kylixs! I am interested in this idea for applying for SoC 2020. Can you guide me how to get started with the repo so that I can start making my proposal?

**Comment by arteevraina** [comment]
Hello @hengyunabc ! Is there a specific channel for this so that I can start discussing the project idea and ask my doubts regarding the` codebase` or should I do it through email?

**Comment by kylixs** [comment]
@arteevraina  The arthas http api branch build step has been added to the content of the issue.
Please join the Arthas discuss group (https://github.com/alibaba/arthas/issues/1198).

**Comment by arteevraina** [comment]
@kylixs Thanks you very much !

**Comment by arteevraina** [comment]
@kylixs @hengyunabc I was following the above mentioned steps but while running `mvn clean package` in `arthas` directory on` http-api `branch. I was getting this build failure. Are there any fixes or workarounds for this?
![image](https://user-images.githubusercontent.com/43968121/83393106-db5f2100-a413-11ea-8ddc-d3d466895e8e.png)

**Comment by kylixs** [comment]
> @kylixs @hengyunabc I was following the above mentioned steps but while running `mvn clean package` in `arthas` directory on`http-api`branch. I was getting this build failure. Are there any fixes or workarounds for this?
> ![image](https://user-images.githubusercontent.com/43968121/83393106-db5f2100-a413-11ea-8ddc-d3d466895e8e.png)

The error details: Failed to delete .../arthas/spy/target/maven-archiver/pom.properties
1) Please stop the Arthas that have been injected;
2) Check what process opened the file spy / target / maven-archiver / pom.properties, please close it. You can use the [Process Hacker](https://processhacker.sourceforge.io/downloads.php) / [Process Explorer ](https://docs.microsoft.com/en-us/sysinternals/downloads/process-explorer) tool to find the file handle to discover the process of opening the file.
3) Or just restart the system and rebuild.

**Comment by arteevraina** [comment]
@kylixs Is `http-api` that displays data in `json format` web ui still in development?

**Comment by kylixs** [comment]
> @kylixs Is `http-api` that displays data in `json format` web ui still in development?

Yes, the first stage of development is completed and is gradually being merged into the master.  
Now you can contribute code, if accepted, it will eventually be merged into the master, but it will take a while.

**Comment by kylixs** [comment]
@sjha2048 Is the proposal ready? Please submit to the official website of SoC 2020.

**Comment by sjha2048** [comment]
@kylixs will do by tomorrow evening.

**Comment by kylixs** [comment]
Some frequently used commands: sm, sc, watch, trace

**Comment by sjha2048** [comment]
Sure, I'll mention these commands.

**Comment by HarshMehta2468** [comment]
@kylixs @hengyunabc  I'm really interested  in contributing to this idea as a part of Asoc 2020. Will you please guide me as to  how should i discuss the project idea so that i can eventually move on towards drafting my proposal?

**Comment by kylixs** [comment]
> @kylixs @hengyunabc I'm really interested in contributing to this idea as a part of Asoc 2020. Will you please guide me as to how should i discuss the project idea so that i can eventually move on towards drafting my proposal?

Please follow the build steps and feedback if you encounter problems.
